File: javalibrary/src/main/java/com/bluthlee/javalibrary/data/BaseBuilder.java
package com.bluthlee.javalibrary.data;

import com.bluthlee.javalibrary.AudioRecorder;

/**
 * Created by LC on 2017/8/31.
 */

public class BaseBuilder {

    protected AudioRecorder.OnFinishListener onFinishListener = null;
    protected String fileName = "";
    protected String dirPath = "";

    public BaseBuilder setOnFinishListener(AudioRecorder.OnFinishListener onFinishListener) {
        this.onFinishListener = onFinishListener;
        return this;
    }

    public BaseBuilder setFileName(String fileName) {
        this.fileName = fileName;
        return this;
    }

    public BaseBuilder setDirPath(String dirPath) {
        this.dirPath = dirPath;
        return this;
    }

}

File: kotlinlibrary/src/main/java/com/bluthlee/library/data/BaseBuilder.kt
package com.bluthlee.library.data

import com.bluthlee.library.AudioRecorder

/**
 * Created by LC on 2017/8/28.
 */
open class BaseBuilder {

    protected var onFinishListener: AudioRecorder.OnFinishListener? = null
    protected var fileName: String = ""
    protected var dirPath: String = ""

    open fun setOnFinishListener(onFinishListener: AudioRecorder.OnFinishListener): BaseBuilder {
        this.onFinishListener = onFinishListener
        return this
    }

    open fun setFileName(fileName: String): BaseBuilder {
        this.fileName = fileName
        return this
    }

    open fun setDirPath(dirPath: String): BaseBuilder {
        this.dirPath = dirPath
        return this
    }

}